File: R/day24.R
is_geometric = function(x){
  if(length(x)<2){ return ( TRUE ) }
  divided=c()
  for(i in 2:length(x)){
    divided = c(divided, x[i]/x[i-1])
  }
  return ( length(unique(divided)) == 1 )
}

print(is_geometric(c(2,3,5))) #FALSE
print(is_geometric(c(2,4,8))) #TRUE
print(is_geometric(0))     #TRUE

File: python/day24.py
def is_geometric(numbers: list) -> bool:
    if len(numbers) < 2:
        return True
    sorted_numbers = sorted(numbers)
    q = sorted_numbers[1]/sorted_numbers[0]
    for i in range(len(numbers)-1):
        if q*sorted_numbers[i] != sorted_numbers[i+1]:
            return False
    return True


assert(is_geometric([1,1,1]) is True)
assert(is_geometric([0.5,0.2]) is True)
assert(is_geometric([4,1,8,16,2]) is True)
assert(is_geometric([1,2,3]) is False)
